gems/pending/spec/postgres_ha_admin/failover_databases_spec.rb

```diff
@@ -1,132 +1,93 @@
 require 'postgres_ha_admin/failover_databases'
-require 'postgres_ha_admin/postgres_ha_admin'
-require 'spec/support/test_env_helper'
 require 'pg'
 
 describe PostgresHaAdmin::FailoverDatabases do
-  subject do
-    described_class.new(TestEnvHelper::CONFIG_DIR, TestEnvHelper::CONFIG_DIR, :dbname => 'travis', :user => 'travis')
-    # described_class.new(TestEnvHelper::CONFIG_DIR, TestEnvHelper::CONFIG_DIR, :dbname => 'vmdb_test')
+  before do
+    @yml_file = Tempfile.new('failover_databases.yml')
+    @logger_file = Tempfile.new('ha_admin.log')
+    @logger = Logger.new(@logger_file)
+    @failover_databases = described_class.new(@yml_file, @logger)
   end
 
-  # let(:connection) { PG::Connection.open(:dbname => 'vmdb_test') }
-  let(:connection) do
-    begin
-      PG::Connection.open(:dbname => 'travis', :user => 'travis')
-    rescue PG::ConnectionBad => _err
-      skip "travis database does not exist"
-    end
+  after do
+    @yml_file.close
+    @yml_file.unlink
+    @logger_file.close
+    @logger_file.unlink
   end
 
-  around(:each) do |example|
-    begin
-      connection.exec("START TRANSACTION")
+  describe "#active_databases" do
+    it "return list of active databases saved in 'config/failover_databases.yml'" do
+      File.write(@yml_file, all_db_list.to_yaml)
+      expect(@failover_databases.active_databases).to contain_exactly(
+        {:type => 'master', :active => true, :host => '1.1.1.1', :user => 'root', :dbname => 'vmdb_test'},
+        {:type => 'standby', :active => true, :host => '2.2.2.2', :user => 'root', :dbname => 'vmdb_test'})
+    end
+  end
 
-      connection.exec(<<-SQL)
-        CREATE TABLE #{described_class::TABLE_NAME} (
+  describe "#update_failover_yml" do
+    before do
+      # @connection = PG::Connection.open(:dbname => 'vmdb_test')
+      begin
+        @connection = PG::Connection.open(:dbname => 'travis', :user => 'travis')
+      rescue PG::ConnectionBad => _err
```

Comment:
You don't need the ` => _err` here. You can just rescue the exception class if you're not interested in the exception instance.